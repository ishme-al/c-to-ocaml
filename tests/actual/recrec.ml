[@@@warning "-26"]
[@@@warning "-27"]
[@@@warning "-32"]
[@@@warning "-69"]

open Core

let rec set_at_index (lst : 'a list) (index : int) (value : 'a) : 'a list =
  match lst with
  | [] -> failwith "Index out of bounds"
  | hd :: tl ->
      if index = 0 then value :: tl else hd :: set_at_index tl (index - 1) value

let rec recursive (n : int) : int =
  printf "recursive(%d)\n" n;
  if Int.( > ) n 1 then recursive (recursive (Int.( - ) n 1)) else n

let () =
  let x : int = recursive 5 in
  exit 0
