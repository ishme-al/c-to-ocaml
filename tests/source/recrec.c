#include <stdio.h>

int recursive(int n) {
    printf("recursive(%d)\n", n);
    if (n > 1) {
        return recursive(recursive(n-1));
    }
    return n;
}

int main() {
    int x = recursive(5);
    return 0;
}