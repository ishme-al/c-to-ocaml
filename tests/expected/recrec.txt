recursive(5)
recursive(4)
recursive(3)
recursive(2)
recursive(1)
recursive(1)
recursive(1)
recursive(1)
recursive(1)
